all/CMakeLists.txt

```diff
@@ -0,0 +1,57 @@
+# This file should be used for development with IDEs for the best discovery
+cmake_minimum_required(VERSION 3.8)
```

Comment:
No, policies are scoped. A new inclusion simply starts a new policy scope. This superbuild CML doesn't use anything 3.8 specific, but that's the lowest modern CMake version I default to in such cases. Every CML must have a `cmake_minimum_required` call immediately followed by a `project` call as the very first thing, otherwise they turn into a spaghetti monster of implicit inter-dependencies.

Reply:
You are unfortunately mistaken on CMake's behavior.

If I have CMake 3.17, and I try to `add_subdirectory` a module with CMake 3.20 as the minimum, it **will error out.**
Logically this makes sense. Different versions change policies and behaviors; how can an older version of CMake possibly know the policy behaviors on the newer one?

This _can_ work if you are using a version of CMake newer than the highest minimum version -- but it will not work on older toolchains (which as I previously mentioned, I am wanting to support)

A simple demonstrative example:

```
$ cmake --version
cmake version 3.17.3

CMake suite maintained and supported by Kitware (kitware.com/cmake).
```


**`CMakeLists.txt`**

```cmake
cmake_minimum_required(VERSION 3.8)

project(Foo)

add_subdirectory(bar)
```

**`bar/CMakeLists.txt`**

```cmake
cmake_minimum_required(VERSION 3.20)

project(Bar)
```

```
$ cmake .
-- Building for: Visual Studio 16 2019
-- Selecting Windows SDK version 10.0.18362.0 to target Windows 10.0.18363.
-- The C compiler identification is MSVC 19.28.29912.0
-- The CXX compiler identification is MSVC 19.28.29912.0
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
-- Check for working C compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe - works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe
-- Check for working CXX compiler: C:/Program Files (x86)/Microsoft Visual Studio/2019/Community/VC/Tools/MSVC/14.28.29910/bin/Hostx64/x64/cl.exe - works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
CMake Error at bar/CMakeLists.txt:1 (cmake_minimum_required):
  CMake 3.20 or higher is required.  You are running version 3.17.3


-- Configuring incomplete, errors occurred!
```

Reply:
Yes, you are correct. However, I prefer to use a version that actually matches the features used in that CML file (as each one should have its own `cmake_minimum_required` anyway), or 3.8, whichever is higher. This is only my preference, you are free to do however.